### ISSUE TITLE
Handling of authentication errors

### ISSUE THREAD

**Issue by lnicola** [agersant/polaris#36]
```
19:40:24 [INFO] rocket::rocket: [~/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket-0.4.2/src/rocket.rs:205] GET /api/browse/:
19:40:24 [INFO] _: [~/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket-0.4.2/src/rocket.rs:289] Matched: GET /api/browse (browse_root)
19:40:24 [INFO] _: [~/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket-0.4.2/src/rocket.rs:297] Outcome: Failure
19:40:24 [WARN] _: [~/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket-0.4.2/src/rocket.rs:318] Responding with 401 Unauthorized catcher.
19:40:24 [INFO] _: [~/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket-0.4.2/src/rocket.rs:120] Response succeeded.
```

I haven't checked, but it might be some sort of CSRF protection. I couldn't find a log out link, so I had to clear my cookies to be able to log in again.

Update: it was Rocket's encrypted cookies.

---

After #41, this no longer happens, but it would be nice if:

- the web UI could detect that it's not authenticating properly and prompt the user to log in again
- there was a way to log out

**Comment by lnicola** [comment]
This is probably because the cookie encryption key is not set. Normally users would be able to set `ROCKET_SECRET_KEY`, but that doesn't work with `rocket::custom`.

**Comment by agersant** [comment]
I think you are correct. The fix here will be to politely ask the Rocket folks to support setting a secret key via rocket::custom, and storing one in the DB on first boot of Polaris so we can re-use the same one every time.

**Comment by lnicola** [comment]
I think it's already possible: https://api.rocket.rs/v0.4/rocket/struct.Config.html#method.set_secret_key.

**Comment by agersant** [comment]
Oh very nice, I think that wasn't there last time I looked. I'll tag this as a bug then, it's on Polaris to fix.

**Comment by lnicola** [comment]
#41 helps, but I wouldn't call this fixed until there's a way to log out. I know that the login page is `/#auth`, but it's not very obvious.

**Comment by agersant** [comment]
I think the ideal experience would be for the client to notice that its session is getting rejected by the server and to show the login page again.

**Comment by lnicola** [comment]
That would be a good idea, but what if you want to log in as a different user? A log out link would still be useful.

**Comment by agersant** [comment]
I agree that would also be valuable.

**Comment by lnicola** [comment]
Actually, this still happens when the session cookie expires.

**Comment by agersant** [comment]
Closing this in favor of two smaller issues:
https://github.com/agersant/polaris-web/issues/38
https://github.com/agersant/polaris/issues/55
